src/main/java/com/linkedin/kmf/apps/MultiClusterMonitor.java

```diff
@@ -70,28 +72,31 @@ public MultiClusterMonitor(Map<String, Object> props, String name) throws Except
 
   @Override
   public void start() {
-    _topicManagementService.start();
-    _produceService.start();
-    _consumeService.start();
-    LOG.info(_name + "/MultiClusterMonitor started");
+    _multiClusterTopicManagementService.start();
+    CompletableFuture<Void> completableFuture = _multiClusterTopicManagementService.topicManagementReady();
+    completableFuture.thenRun(() -> {
+      _produceService.start();
+      _consumeService.start();
+    });
+    LOG.info(_name + "/MultiClusterMonitor started.");
```

Comment:
I would put the log line inside `.thenRun` lambda. Secondly, if `producerService.start` throws, should `consumerService` start? Perhaps not. Consider using `CompletableFuture.handle` to handle exceptions.

Reply:
Sure. Lambda refactoring is already in the backlog, since the constructors for producers and consumers are too large in general as well.